api/metrics-api-jaxrs-1.1/src/main/java/org/hawkular/metrics/api/jaxrs/handler/MetricHandler.java

```diff
@@ -82,6 +88,41 @@
     @HeaderParam(TENANT_HEADER_NAME)
     private String tenantId;
 
+    @POST
+    @Path("/")
+    @ApiOperation(value = "Create metric definition.", notes = "Clients are not required to explicitly create "
+            + "a metric before storing data. Doing so however allows clients to prevent naming collisions and to "
+            + "specify tags and data retention.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Metric definition created successfully"),
+            @ApiResponse(code = 400, message = "Missing or invalid payload", response = ApiError.class),
+            @ApiResponse(code = 409, message = "Metric with given id already exists",
+                    response = ApiError.class),
+            @ApiResponse(code = 500, message = "Metric definition creation failed due to an unexpected error",
+                    response = ApiError.class)
+    })
+    public Response createMetric(
+            @ApiParam(required = true) MetricDefinition metricDefinition,
+            @Context UriInfo uriInfo
+    ) {
+        MetricId<?> id = new MetricId(tenantId, metricDefinition.getType(), metricDefinition.getId());
+        Metric<?> metric = new Metric<>(id, metricDefinition.getTags(), metricDefinition.getDataRetention());
```

Comment:
It's not better readability, it's better syntax highlighting in your 
IDE. Anyway, leave it like this if you want.